src/vivarium/framework/state_machine.py

```diff
@@ -236,6 +239,20 @@ def __init__(
     # Public methods #
     ##################
 
+    def is_initial_state(self) -> bool:
+        """Determines if simulants could be initialized into this state.
+
+        Note: this will incorrectly return True if initialization_weights is a
```

Comment:
This kinda seems like an important bug we shouldn't introduce. Is this case handled downstream at all?

Reply:
The only implication of this happening is that we will throw a validation error if you specify an `initial_state` for your model if you specify a callable or artifact key for any state in that model even if that callable will always return 0, or if the key in the artifact only contains 0s. This doesn't seem like a big deal to me.

Reply:
I'd recommend at least pointing out in the docstring that it's not a big deal